scio-extra/src/main/scala/com/spotify/scio/extra/sorter/syntax/SCollectionSyntax.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.extra.sorter.syntax
+
+import java.lang.{Iterable => JIterable}
+
+import com.spotify.scio.annotations.experimental
+import com.spotify.scio.coders.{Coder, CoderMaterializer}
+import com.spotify.scio.extra.sorter.SortingKey
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.coders.{IterableCoder, KvCoder}
+import org.apache.beam.sdk.extensions.sorter.{BufferedExternalSorter, SortValues}
+import org.apache.beam.sdk.values.KV
+
+import scala.collection.JavaConverters._
+
+final class SorterOps[K1, K2: SortingKey, V](self: SCollection[(K1, Iterable[(K2, V)])]) {
+
+  /**
+   * Takes an [[SCollection]] with elements consisting of a primary key and iterables
+   * over (secondary key, value) pairs, and returns an [[SCollection]] of the same elements
+   * but with values sorted lexicographicly by the secondary key.
+   *
+   * The secondary key needs to be encoded as a [[String]] or [[Array[Byte]].
+   * [[SortValues]] compares bytes lexicographically and may write secondary key-value pairs to disk.
+   *
+   * @note The primary key is explicit here only because this
+   * transform is typically used on a result of a [[PairSCollectionFunctions.groupByKey]].
+   *
+   * @param sorterOptions see [[BufferedExternalSorter.Options]]
+   */
+  @experimental
+  def sortValues(sorterOptions: BufferedExternalSorter.Options)(
```

Comment:
Do we want a default options? IMO the Beam default Hadoop `SequenceFile` sorter is 2x slower than the native sorter. The only other option is memory. Should we just have a `memoryMB: Int = 128`?

Reply:
to be honest , not sure not that keen in setting defaults here. I guess i would prefer users to look at the options and set them according to their use case?

Reply:
At least make native sorter a default and exclude Hadoop dependency? The only reason it's the default in Beam is some review prefer it for "backwards compatibility" ;)

We can then expose `memoryMB` as a required option?

Reply:
me and @nevillelyh talked in IRL; and @nevillelyh did some benchmarks on the default `SorterType.HADOOP` and concluded that is slower then the `SorterType.NATIVE`. With this in mind we will restrict this to the later and only expose `memoryMB `